util/topgen/lib.py

```diff
@@ -495,3 +495,15 @@ def make_bit_concatenation(sig_name: str,
         acc += [',', item_indent, item]
     acc += ['\n', ' ' * end_indent, '}']
     return ''.join(acc)
+
+
+def param_value (ip_param, module_params_decl):
+    '''Return param value of an exposed parameter.
+       If there are no explicitly declared value under module, use default.
+       If there is an explicitly declared value, use that instead.
+
+    '''
+    if (ip_param["name"] in module_params_decl.keys()):
```

Comment:
Nit: You don't need the outer parens here. You can also avoid the multiple return statements with something like this (which also avoids double lookups, not that we really care about the performance impact):
```python3
    mod_val = module_params_decl.get(ip_param["name"])
    return mod_val if mod_val is not None else ip_param["default"]
```

Reply:
thanks! i didnt even know there was a `get` method, that's super helpful!